Apps/Patient/src/Services/PatientDataService.cs

```diff
@@ -15,63 +15,119 @@
     {
         private readonly IPatientDataRepository patientDataRepository;
         private readonly IPersonalAccountsService personalAccountsService;
+        private readonly IMapper mapper;
 
-        public PatientDataService(IPatientDataRepository patientDataRepository, IPersonalAccountsService personalAccountsService)
+        public PatientDataService(IPatientDataRepository patientDataRepository, IPersonalAccountsService personalAccountsService, IMapper mapper)
         {
             this.patientDataRepository = patientDataRepository;
             this.personalAccountsService = personalAccountsService;
+            this.mapper = mapper;
         }
 
         public async Task<PatientDataResponse> Query(PatientDataQuery query, CancellationToken ct)
         {
-            var pid = await this.ResolvePidFromHdid(query.Hdid).ConfigureAwait(true);
-            var results = await this.patientDataRepository.Query(
-                new HealthServicesQuery(pid, query.PatientDataTypes.Select(this.MapToHealthServiceCategory)),
-                ct).ConfigureAwait(true);
+            Guid pid = await this.ResolvePidFromHdid(query.Hdid).ConfigureAwait(true);
+            List<Task<PatientDataQueryResult>> tasks = new();
+            tasks.Add(this.HandleServiceQuery(pid, query, ct));
+            tasks.Add(this.HandleDataQuery(pid, query, ct));
 
-            return new PatientDataResponse(results.Items.Select(this.MapToPatientData).ToArray());
+            IEnumerable<PatientDataQueryResult> results = await Task.WhenAll(tasks).ConfigureAwait(true);
+            return new PatientDataResponse(results.SelectMany(r => r.Items).Select(this.MapToPatientData).ToArray());
         }
 
         public async Task<PatientFileResponse?> Query(PatientFileQuery query, CancellationToken ct)
         {
-            var pid = await this.ResolvePidFromHdid(query.Hdid).ConfigureAwait(true);
-            var file = (await this.patientDataRepository.Query(new PatientDataAccess.PatientFileQuery(pid, query.FileId), ct).ConfigureAwait(true)).Items.FirstOrDefault() as PatientFile;
+            Guid pid = await this.ResolvePidFromHdid(query.Hdid).ConfigureAwait(true);
+            PatientFile? file = (await this.patientDataRepository.Query(new PatientDataAccess.PatientFileQuery(pid, query.FileId), ct).ConfigureAwait(true)).Items.FirstOrDefault() as PatientFile;
 
             return file != null
                 ? new PatientFileResponse(file.Content, file.ContentType)
                 : null;
         }
 
-        private async Task<Guid> ResolvePidFromHdid(string patientHdid) =>
-            (await this.personalAccountsService.GetPatientAccountAsync(patientHdid).ConfigureAwait(true)).PatientIdentity.Pid;
+        private static IEnumerable<HealthServiceCategory> ExtractToHealthServiceCategoryArray(IEnumerable<PatientDataType> patientDataType)
+        {
+            foreach (PatientDataType pdt in patientDataType)
```

Comment:
Could this be simplified using linq instead? 

Example: "return patientDataType.Where(x => x == PatientDataType.OrganDonorRegistrationStatus).Select(_ => HealthServiceCategory.OrganDonor);"

Reply:
Very true this was generated by Copilot and it worked so I didn't think much of it.